Makefile

```diff
@@ -359,6 +360,15 @@ ifeq ($(CAPSTONE_SHARED),yes)
 endif
 endif
 
+$(SUPPORT):
+	echo '#ifndef CAPSTONE_SUPPORT_H' > $@
+	echo '#define CAPSTONE_SUPPORT_H' >> $@
+	for a in $(CAPSTONE_ARCHS) ; do \
+		ARCH=$$(echo $$a|tr a-z A-Z); \
+		echo '#define CS_SUPPORT_ARCH_'$${ARCH} 1 >> $@ ; \
+	done
+	echo '#endif' >> $@
```

Comment:
You should add the API version too, encoded as a flat integer like #define CAPSTONE_VERSION(a,b,c) ((a)<<24+(b)<<16+(c)<<8)